### DISCUSSION TITLE
Using Yolov5 in cloud function

### DISCUSSION THREAD

**Discussion by maryamWOW** [ultralytics/yolov5#9789] [Q&A]
I am trying to run yolov5 in cloud function and I am using the following version for torch and torchvision but I get error while triggering my function:

requirement.txt
torchvision>=0.8.1
https://download.pytorch.org/whl/cpu/torch-1.10.2%2Bcpu-cp39-cp39-linux_x86_64.whl 

Error:
 RuntimeError: Couldn't load custom C++ ops. This can happen if your PyTorch and torchvision versions are incompatible, or if you had errors while compiling torchvision from source. For further information on the compatible versions, check https://github.com/pytorch/vision#installation for the compatibility matrix.

Would you please help me how can I fix this issue? 

torch.version:1.10.2+cpu
torchvision.version:0.11.3+cu102

**Comment by glenn-jocher** [comment]
See https://pytorch.org/get-started/locally/ and https://pypi.org/project/torchvision/ for torchvision compatibility table

**Reply by maryamWOW** [reply]
Thanks, Does it need any requirements for running on cloud function? I can deploy yolov5 on cloud function, but after triggering the function and sending a video , I get incompatible packages error.

**Reply by glenn-jocher** [reply]
@maryamWOW YOLOv5 requirements are shown in requirements.txt:
https://github.com/ultralytics/yolov5/blob/master/requirements.txt

**Reply by maryamWOW** [reply]
I tested with that requirements, and after deploying cloud function, I got the following errors, then after changing torch vision(as mentioned above) , it is deployed, however , in runtime execution, I got torch compatibility error.



Deployment failure:
Build failed: Collecting matplotlib>=3.2.2
Using cached matplotlib-3.6.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (9.4 MB)
Collecting numpy>=1.18.5
Using cached numpy-1.23.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
Collecting opencv-python>=4.1.1
Using cached opencv_python-4.6.0.66-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (60.9 MB)
Collecting Pillow>=7.1.2
Using cached Pillow-9.2.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
Collecting PyYAML>=5.3.1
Using cached PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (701 kB)
Collecting requests>=2.23.0
Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting scipy>=1.4.1
Using cached scipy-1.9.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (33.8 MB)
Collecting torch>=1.7.0; Error ID: 0ea8a540
